### PR TITLE
fix: added role for href button

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

Added a check on href manageAnchor() method to make sure role attribute is set to "link" when there is an href attribute in sp-button
## Related issue(s)

- #3245

## Motivation and context

When href is added to sp-button, a link with aria-hidden is rendered. And at the same time, role is also missing on the component. So effectively, the interactive nature of component is not communicated to screen reader users.
## How has this been tested?
In Documentation site as well as in Storybook inspecting the shadowRoot

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Screenshots (if appropriate)
Screenshot 2023-06-20 at 1 27 41 PM

Documentation:

Screenshot 2023-06-21 at 3 36 22 PM

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Review by najikahalsema — CHANGES_REQUESTED** [review]
A few things:

I checked on my end that the role was added. When it comes to accessibility and roles displaying properly, we have to do a bit more work than just going into the shadowRoot, because sometimes things seem okay from the DOM Tree, but don't actually show up in the screen reader for one reason or another. 
The first thing we want to do to confirm everything is OK is to view the accessibility tree in the chrome dev tools. This will tell us about the attributes that do or don't apply to an element. This screenshot is of the DOM tree from your branch, so it looks good and the role does display. 
image

The second thing to check, and is perhaps more important than just looking at the a11y tree, is to actually utilise your computer's voiceover tool to test the behaviour from the perspective of a user who would be using a screenreader. On a Mac, we can find this in Settings -> Accessibility --> VoiceOver. There's an option to pin the accessibility controls to the menu bar, which I would recommend doing. We can chat about how to use the voiceover in a meeting if you'd like more information because it's kind of hard to explain what to do through a github comment. 

Next, I noticed in the related issue that we should probably add some documentation about the href attribute since it's not documented. We don't explicitly refer to an `href` property in the ButtonBase, so adding the attribute via JSdoc isn't an option. I would recommend adding a section in the documentation for Button, under the "Handling Events" section, that says something along the lines of:
> in addition to handling events like a native `<button>` HTML element, one can also use a Button like the `<a>` HTML Element by using the `href` and optional `_target` attributes.

I would also include a quick HTML example similar to what's already in the docs.

**Comment by Rajdeepc** [comment]
@najikahalsema  Updated documentation. Please have a look and let me know if you need any changes on this. Thank you.

File: packages/button/README.md [hunk 1]
```diff
@@ -306,6 +306,17 @@ Events handlers for clicks and other user actions can be registered on a
 <sp-button onclick="spAlert(this, '<sp-button> clicked!')">Click me</sp-button>
 ```
 
+In addition to handling events like a native `<button>` HTML element, one can also use a Button like the `<a>` HTML Element by using the href and optional \_target attributes.
```
[the remark this answers is not in the document]
Comment (najikahalsema) [reply packages/button/README.md:309]: "In place *of", not "in place on" 😉

[the remark this answers is not in the document]
Comment (Rajdeepc) [reply packages/button/README.md:309]: damn i just copy pasted :(

**Review by najikahalsema — APPROVED** [review]
Looks great!

**Pull request merged**